### ISSUE TITLE
`Path` specs should be implicitly unlocked

### ISSUE THREAD

**Issue by esttorhe** [CocoaPods/CocoaPods#3764]
I’m trying to do `pod try` on `DFBehaviors` but I’m getting this error

```console
Fetching podspec for `DFBehaviors` from `../`
[!] Unable to satisfy the following requirements:

- `DFBehaviors (from `../`)` required by `Podfile`
- `DFBehaviors (from `../`)` required by `Podfile`
- `DFBehaviors (= 0.2.3)` required by `Podfile.lock`
```

**Comment by segiddins** [comment]
```diff
diff --git a/lib/cocoapods/installer/analyzer.rb b/lib/cocoapods/installer/analyzer.rb
index 29df827..b70f37f 100644
--- a/lib/cocoapods/installer/analyzer.rb
+++ b/lib/cocoapods/installer/analyzer.rb
@@ -397,6 +397,7 @@ module Pod
         else
           pods_to_update = result.podfile_state.changed + result.podfile_state.deleted
           pods_to_update += update[:pods] if update_mode == :selected
+          pods_to_update += podfile.dependencies.select(&:local).map
           LockingDependencyAnalyzer.generate_version_locking_dependencies(lockfile, pods_to_update)
         end
       end
```

**Comment by segiddins** [comment]
Closed via 82edf68.
